app/services/db/establish_connection/read_replica.rb

```diff
@@ -0,0 +1,18 @@
+module Db
+  module EstablishConnection
+    class ReadReplica
+      def self.call
+        rails_env = Rails.env
+        return if rails_env.test?
+        env = Figaro.env
+        ActiveRecord::Base.establish_connection(
+          adapter: 'postgresql',
+          database: rails_env.production? ? env.database_name : "upaya_#{rails_env}",
+          host: env.database_read_replica_host,
```

Comment:
the two classes seem to only differ by this one line...did you consider having two methods on a single class? Like `Db::ConnectionManager.in_read_replica` and `.in_leader_node` or something?

Reply:
For services classes we decided to only ever have one method `call`

Reply:
Ok! Do we have documentation on that convention so I can read more?

Does `call` need to be a no-args method? Ex could it take in a block as an argument, so that callers don't need to remember to clean up afterwards?

Reply:
It can take args.  It was Moncef that championed it and I actually thought it was a good idea.  No docs but I can probably dig up some threads

Reply:
I went ahead and switched this up to use a block